src/components/ScheduleList.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {useEffect, useState} from 'react';
+import styled from 'styled-components';
+import dayjs from 'dayjs';
+import {GroupedVirtuoso} from 'react-virtuoso';
+import ScheduleCard from './ScheduleCard';
+
+const List = styled.ul`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  margin: 20px 0 0;
+  padding: 0 20px 0;
+  overflow-x: scroll;
+`;
+
+const DayTitle = styled.p`
+  font-weight: 600;
+  font-size: 16px;
+  padding-top: 60px;
+  padding-bottom: 30px;
+  background: #e4e4e4;
+  margin: 0;
+`;
+
+interface IProps {
+  items: any;
+}
+
+interface GroupDataReturn {
+  groups: any[];
+  groupsData: any[];
+}
+
+interface GeneratedGroupData extends GroupDataReturn {
+  groupsCount: any[];
+}
+
+interface GroupWithTitle {
+  title: string;
+}
+
+function groupData(data: any[]): GroupDataReturn {
+  const result: GroupDataReturn = {
+    groups: [],
+    groupsData: [],
+  };
+
+  if (!data) {
+    return result;
+  }
+
+  data.map((item: GroupWithTitle | any) => {
+    if (item.title) {
+      result.groups = [...result.groups, item];
+    } else {
+      result.groupsData = [...result.groupsData, item];
+    }
+  });
```

Comment:
If the return value of a `.map` isn't used, maybe a `for (const x of [])` would better signify the code's intent